storegateway: check external labels when set out_of_order for stats

narqo opened the pull request:
#### What this PR does

We've noticed that the store-gateway sometimes reports confusing data in its `cortex_bucket_store_series_blocks_queried`. After looking closer, it has turned out that the `out_of_order` label is derived from the "from-out-of-order" hint. @bboreham mentioned that, after the compaction, blocks don't preserve the hint. Instead, Mimir sets the `__out_of_order__` external label. 

This PR updates the internal `blockQueriedMeta` to take this into account.

tacole02 reviewed (commented):
Changelog LGTM

dimitarvdimitrov commented on pkg/storegateway/stats.go:
```diff
@@ -110,6 +111,11 @@ func newBlockQueriedMeta(meta *block.Meta) blockQueriedMeta {
+	// The compactor doesn't preserve the "FromOutOfOrder" hints. Instead, we should check for the external label, that Mimir sets.
+	if !m.outOfOrder && meta.Thanos.Labels != nil {
+		m.outOfOrder = meta.Thanos.Labels[mimir_tsdb.OutOfOrderExternalLabel] == mimir_tsdb.OutOfOrderExternalLabelValue
+	}
```
should this be something like a method on the block.meta? i can see it being confusing to other people too

56quarters commented on pkg/storegateway/stats.go:
```diff
@@ -110,6 +111,11 @@ func newBlockQueriedMeta(meta *block.Meta) blockQueriedMeta {
 		m.level = "unknown/old_block"
 	}
+
+	// The compactor doesn't preserve the "FromOutOfOrder" hints. Instead, we should check for the external label, that Mimir sets.
```
Can this logic live in `block.Meta`? Maybe add a method to `block.Meta` that checks both if the compactor set out-of-order or if the external labels have it? It seems to be entirely determined from block metadata.

narqo commented:
@56quarters @dimitarvdimitrov you commented about the same thing. I've addressed the comment, extracting the functionality into a method in the `block.Meta`. This came with a big chunk of unexpected changes all over the place:

That is, the `storage/tsdb` imported `storage/tsdb/block`, but the `storage/tsdb/block.Meta` now needed constants from `storage/tsdb`. To resolve the importing cycle, I moved the constants to `storage/tsdb/block`.

PTAL.
